### PR TITLE
[7.9.x] chore: bump 7.9.0 to 7.9.1

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It bumps the versions for the stack and the agent, to use the latest 7.9.1 version

## Why is it important?
The 7.9.0 binaries for the elastic-agent dissapeared (https://artifacts-api.elastic.co/v1/search/7.9.0/elastic-agent returns a 404 not-found error), causing all tests to fail.

```log
[2020-09-14T07:49:39.868Z] time="2020-09-14T07:49:39Z" level=warning msg="The Elastic artifacts API is not available yet" arch=x86_64 artifact=elastic-agent elapsedTime=27.858337442s error="GET request failed with 404" extension=rpm os=linux retry=10 statusEndpoint="https://artifacts-api.elastic.co/v1/search/7.9.0/elastic-agent" version=7.9.0
```

Bumping the stack and the agent to 7.9.1 brings back the tests again.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [x] Run the CI with the 7.9.1 parameters

### CODE REVIEW

**Review by v1v — APPROVED** [review]

File: e2e/_suites/ingest-manager/README.md [hunk 1]
```diff
@@ -45,10 +45,10 @@ This is an example of the optional configuration:
 
    ```shell
    # There should be a Docker image for the runtime dependencies (elasticsearch, kibana, package registry)
-   export STACK_VERSION=7.9.0
+   export STACK_VERSION=7.9.1
    # (Fleet mode) This environment variable will use a fixed version of the Elastic agent binary, obtained from
-   # https://artifacts-api.elastic.co/v1/search/7.9.0/elastic-agent
-   export ELASTIC_AGENT_DOWNLOAD_URL="https://snapshots.elastic.co/7.9.0-81c43e7b/downloads/beats/elastic-agent/elastic-agent-7.9.0-linux-x86_64.tar.gz"
+   # https://artifacts-api.elastic.co/v1/search/7.9.1/elastic-agent
+   export ELASTIC_AGENT_DOWNLOAD_URL="https://staging.elastic.co/7.9.1-7b774025/downloads/beats/elastic-agent/elastic-agent-7.9.1-linux-x86_64.tar.gz"
```
Comment (ruflin) [inline e2e/_suites/ingest-manager/README.md:51]: As 7.9.1. is out, I wonder why we use staging here? Couldn't we just use the stable release for the 7.9 testing?

Or if we want to test the snapshot, then it should probably be something like `docker.elastic.co/beats/elastic-agent:7.9.2-SNAPSHOT`? But this is not avaiable under artifacts yet (perhaps something to be changed?)

**Review by cachedout — APPROVED** [review]

File: e2e/_suites/ingest-manager/README.md (hunk 1 above)
Comment (mdelapenya) [reply e2e/_suites/ingest-manager/README.md:51]: If we query `https://artifacts-api.elastic.co/v1/search/7.9.1/elastic-agent` it returns that `staging` URL, and that has recently changed.

Related to 7.9.2, I'd ask: what versions do we want to test in the 7.9.x maintenance branch?

Comment (EricDavisX) [reply e2e/_suites/ingest-manager/README.md:51]: from my perspective... I'd say, we kinda rather never plan to run a test against a GA version.  As funny as that may seem.  @ruflin - we'd always want to set the tests to pick up the latest code available from a given release line for all components (agent, kibana, es, relating-registry if applicable), the snapshots are that right?   

The question of what artifact registry is beyond me... things change and we will keep up with 'current'.  I was concerned at first that the 7.9.0 artifact was 'gone'... but if this is just an internal access point I guess that's ok and fine.

Syncing back to your question @mdelapenya - do we not have a way to dynamically leave the 7.9.x branch to pull the latest artifacts / snapshots available for 7.9.x instead of hard-coding it to a specific staging hash?  Let me look this up myself...

**Comment by EricDavisX** [comment]
ok, i had a chat with Manu and we decided (unless anyone disagrees) that we will provide support for nightly runs of the master (currently 8.0) and previous major release (7.x currently).  These should always have full snapshot (or staging I guess) urls of all components we can pull.  We will provide ad-hoc support for running the patch version tests, but they may also require a pr to update the agent version so it can be found correctly during test run.
 - if the Agent has no changes in the x.x.x patch release, it binaries and Docker image for it won't exist - we can ask Infra to build this or just go with our plan here of manually setting the version to the correct latest one before running the test.  This seems like the less confusing path and we can avoid spending-time on patch release tests that often don't relate to the code we are covering here.  Note that the patch release changes for Agent will be covered during the PR CI, once we get that done (imminently!!!).  So we will have coverage.

@mdelapenya did I forget anything we want to mention?  I believe it means that we can use this PR as is and move forwards.  @ruflin let us know if you are ok with this approach or if we should talk to Infra about building all components daily for minor release .x versions

**Comment by mdelapenya** [comment]
Yes, we will bump the minor versions _after finding issues/failures_ in these tests.

Because the CI provides full capabilities to run multiple combinations of versions, we will delegate the verification of certain corner cases to manual triggers of the jobs with the proper values. We have created an issue to document how to run those manual regression tests here: #287

**Comment by mdelapenya** [comment]
Merging as discussed above, thanks!

**Comment by ruflin** [comment]
@EricDavisX @mdelapenya Thanks for pushing this forward. It might be worth to write down somewhere on what we test with which versions as I'm pretty sure, this question will bubble up again.

**Comment by mdelapenya** [comment]
> @EricDavisX @mdelapenya Thanks for pushing this forward. It might be worth to write down somewhere on what we test with which versions as I'm pretty sure, this question will bubble up again.

Yes, we can document in the repo what expected t be tested or each branch. Maybe main README for the ingest-manager test suite is a good place to start.
